operator/version/version.go

```diff
@@ -15,13 +15,17 @@
 package version
 
 import (
+	"time"
+
 	pkgversion "istio.io/istio/operator/pkg/version"
 	buildversion "istio.io/pkg/version"
 )
 
 const (
 	// OperatorCodeBaseVersion is the version string from the code base.
 	OperatorCodeBaseVersion = "1.17.0"
+	OperatorEOLYear         = 2023
+	OperatorEOLMonth        = time.September
```

Comment:
If I look at https://istio.io/latest/docs/releases/supported-releases/ and go back to the release 1 yr ago, it was 1.13 and it's EOL was Oct 12. Thus I would suspect that we don't want to actually output a warning until November of 2023.

Reply:
Actually, this is configurable for each release, I am not sure 1.17 when will be EOL, so I just put that date.

Reply:
1.18.0 which is now what the main branch is, will likely EOL in 10/23 so maybe make this November.